src/Surfnet/StepupGateway/GatewayBundle/Entity/SamlEntity.php

```diff
@@ -110,6 +109,16 @@ public function toServiceProvider()
         $configuration['entityId']             = $this->entityId;
         $configuration['configuredLoas']       = $decodedConfiguration['loa'];
 
+        $configuration['secondFactorOnly'] = false;
+        if (isset($decodedConfiguration['second_factor_only'])) {
+            $configuration['secondFactorOnly'] = (bool) $decodedConfiguration['second_factor_only'];
```

Comment:
The default is great. I just mean this:

```diff
    $configuration['secondFactorOnly'] = false;
    if (isset($decodedConfiguration['second_factor_only'])) {
-       $configuration['secondFactorOnly'] = (bool) $decodedConfiguration['second_factor_only'];
+       $configuration['secondFactorOnly'] = $decodedConfiguration['second_factor_only'];
    }
```

When it's set, there must be a boolean, so let's just take that value.